chiware/stats.go

```diff
@@ -12,6 +12,10 @@ import (
 	"github.com/rudderlabs/rudder-go-kit/stats"
 )
 
+const (
+	UnknownPath = "/unknown"
+)
+
 func StatMiddleware(ctx context.Context, router chi.Router, s stats.Stats, component string) func(http.Handler) http.Handler {
```

Comment:
By default `StatMiddleware` shall redact unknown paths, but we should also let the user choose to override this behaviour if needed through an option, e.g. using `WithRedactUnknownPaths(false)`
```suggestion
type config struct {
	redactUnknownPaths bool
}

type Option func(*config)

// RedactUnknownPaths sets the redactUnknownPaths flag.
// If set to true, the path will be redacted if the route is not found.
// If set to false, the path will be used as is.
func RedactUnknownPaths(redactUnknownPaths bool) Option {
	return func(c *config) {
		c.redactUnknownPaths = redactUnknownPaths
	}
}

func StatMiddleware(ctx context.Context, router chi.Router, s stats.Stats, component string, options ...Option) func(http.Handler) http.Handler {
	conf := config{
		redactUnknownPaths: true,
	}
	for _, option := range options {
		option(&conf)
	}
```

Reply:
makes sense, Modified.